includes/Checker/Checks/Abstract_PHP_CodeSniffer_Check.php

```diff
@@ -75,6 +75,23 @@ final public function run( Check_Result $result ) {
 		// Backup the original command line arguments.
 		$orig_cmd_args = $_SERVER['argv'];
 
+		if ( class_exists( '\PHP_CodeSniffer\Config' ) ) {
+			/*
+			 * Reset \PHP_CodeSniffer\Config::$overriddenDefaults to prevent
+			 * incorrect results when running multiple checks.
+			 *
+			 * PHPStan ignore reason: PHPStan raised an issue because we can't
+			 * use class in ReflectionClass.
+			 *
+			 * @phpstan-ignore-next-line
+			 */
+			$reflected_phpcs_config = new \ReflectionClass( '\PHP_CodeSniffer\Config' );
+			$overridden_defaults    = $reflected_phpcs_config->getProperty( 'overriddenDefaults' );
+			$overridden_defaults->setAccessible( true );
+			$overridden_defaults->setValue( array() );
+			$overridden_defaults->setAccessible( false );
+		}
```

Comment:
@mukeshpanchal27 How does moving this here fix the fatal error? Related is my question in https://github.com/10up/plugin-check/issues/210#issuecomment-1611792431, I'm still unclear what the actual error is. That may help answer my question here.

Just from looking at it, I am a bit confused why we need to copy logic that only applied to tests over to production code. Can you clarify why that is needed?

Reply:
Apologies, @felixarntz, for forgetting to mention the updates I made in the PR description. Could you please take a moment to review the changes and let me know if they provide helpful information for the PR review?

Reply:
Thanks @mukeshpanchal27, it's clear to me now!